sirepo/template/activait.py

```diff
@@ -1228,103 +1218,79 @@ def _x_y(self, run_dir=None):
+        return self.file[self.io.input.path], o, numpy.array([])
 
     def _gen_image(self):
         import matplotlib.pyplot as plt
 
-        if (
-            _param_to_image(self.info)
-            and not self.data.args.method in _POST_TRAINING_PLOTS
-        ):
-            for section in ("top", "right", "bottom", "left"):
-                self.axes[self.row, 0].spines[section].set_visible(False)
-            self.axes[self.row, 0].text(
-                0.2,
-                0.2,
-                "Params:\n" + ", ".join([str(round(n, 3)) for n in self.input]),
-                style="italic",
-                fontsize=10,
-            )
-            self.axes[self.row, 1].imshow(self.output)
-            return
-        if _image_out(self.info):
-            c = [self.input, self.output]
-            if self.original is not None:
-                c.insert(0, self.original)
-            for i, column in enumerate(c):
-                self.axes[self.row, i].imshow(column)
-            return
-        plt.subplot(_IMG_ROWS, _IMG_COLS, self.row + 1)
         plt.xticks([])
         plt.yticks([])
-        plt.imshow(self.input)
-        if len(self.file[self.io.output.path].shape) == 1:
-            if "label_path" in self.io.output:
-                plt.xlabel(
-                    pkcompat.from_bytes(
-                        self.file[self.io.output.label_path][self.output]
-                    )
+        plt.imshow(self.currentImage)
+
+    def _append_input_image(self, data, index):
+        self.currentImage = data[index]
+        if self.io.input.kind == "f":
+            self.currentImage = self.currentImage.astype(float)
+        if self.currentImage.ndim == 1:
+            self.xIsParameters = True
+            self.inputs.append(self.currentImage)
+            return
+        self._gen_image()
+        self.inputs.append(self._pyplot_data_url())
+
+    def _append_output_image(self, data, index):
+        if data.ndim == 1 and "label_path" in self.io.output:
+            self.imageToLabels = True
+            self.outputs.append(pkcompat.from_bytes(
+                    self.file[self.io.output.label_path][data[index]]
                 )
-            else:
-                plt.xlabel(self.output)
-        else:
-            plt.xlabel("\n".join([str(l) for l in self.output]))
+            )
+            return
+        self.currentImage = data[index]
+        self._gen_image()
+        self.outputs.append(self._pyplot_data_url())
+
+    def _append_original_image(self, data, index):
+        if data.size != 0:
+            self.currentImage = data[index]
+            self._gen_image()
+            self.originals.append(self._pyplot_data_url())
+
+    def _final_images(self):
+        res = PKDict(
+            paramToImage=_param_to_image(self.info),
+            xIsParameters=self.xIsParameters,
+            imageToLabels=self.imageToLabels,
+            pred=False,
+            x=self.inputs,
+            y=self.outputs,
```

Comment:
Can `x` and `y`  be made more descriptive? I'm not sure what they represent, especially on the javascript side where the context here is lost.